File: Code/apps/problems_directory/problems/reverse-string/drivers/javascript/driver.js
const fs = require('fs');
const DELIMITER = '$$$DELIMITER$$$';

try {
    const inputData = fs.readFileSync('/app/input.txt', 'utf-8');
    // Split by delimiter and filter empty entries
    const testCases = inputData.split(DELIMITER).map(tc => tc.trim()).filter(tc => tc !== '');

    testCases.forEach((testCase) => {
        const lines = testCase.split('\n').map(l => l.trim()).filter(l => l !== '');
        
        let arg0 = lines[0].trim();
        if (arg0.startsWith('"') && arg0.endsWith('"')) {
            arg0 = arg0.slice(1, -1);
        }

        const result = reverseString(arg0);
        
        // Output JSON stringified result
        console.log(JSON.stringify(result));
        console.log(DELIMITER);
    });
} catch (err) {
    console.error(err);
}

File: Code/apps/problems_directory/problems/reverse-string/drivers/python/driver.py
import sys
import json

DELIMITER = "$$$DELIMITER$$$"

def main():
    try:
        with open("/app/input.txt", "r") as f:
            input_data = f.read()
            
        test_cases = input_data.split(DELIMITER)
        sol = Solution()
        
        for test_case in test_cases:
            if not test_case.strip(): continue
            lines = [line.strip() for line in test_case.strip().split('\n') if line.strip()]
            
            arg0 = lines[0].strip().strip('"')

            result = sol.reverseString(arg0)
            
        
            
            print(json.dumps(result, separators=(',', ':')))
            print(DELIMITER)
            
    except Exception as e:
        sys.stderr.write(str(e))

if __name__ == "__main__":
    main()